src/routing_table/prefix.rs

```diff
@@ -224,18 +226,19 @@ impl<T: Clone + Copy + Default + Binary + Xorable> Debug for Prefix<T> {
 }
 
 #[cfg(test)]
-impl Prefix<u8> {
-    pub fn from_str(bits: &str) -> Prefix<u8> {
+impl FromStr for Prefix<u8> {
+    type Err = String;
+    fn from_str(bits: &str) -> Result<Prefix<u8>, String> {
         let mut name = 0u8;
         for (i, bit) in bits.chars().enumerate() {
             if bit == '1' {
                 name |= 1 << (7 - i);
             } else if bit != '0' {
-                panic!("'{}' not allowed - the string must represent a binary number.",
-                       bit);
+                return Err(format!("'{}' not allowed - the string must represent a binary number.",
+                                   bit));
```

Comment:
Clippy didn't like us naming that `from_str()` without using the trait.  The trait requires a `Result` be returned.